### PR TITLE
interface addition and correction

### PR DESCRIPTION
**Pull request by ustulation**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/495)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/structured_data.rs [hunk 1]
```diff
@@ -57,6 +57,19 @@ impl StructuredData {
                    previous_owner_signatures: previous_owner_signatures
                  }
     }
+
+    /// This is a static function required for computing a name give the tag-type and identifier to
+    /// be use by GETs
+    pub fn compute_name(type_tag: u64, identifier: &NameType) -> NameType {
```
Comment (benjaminbollen) [inline src/structured_data.rs]: What is the use-case for having compute_name as a public function, when we already have pub fn name?

Comment (benjaminbollen) [inline src/structured_data.rs]: Ok I see your use-case

Comment (benjaminbollen) [inline src/structured_data.rs]: Never mind my comments

Comment (ustulation) [inline src/structured_data.rs]: Explanation's in the documentaion :) .. Here it is in detail: In GETs I only have the type_tag and identifier, and i have to get the location by doing magic on it. That magic is the way structureddata name is calculated which also maintains an invariant. It needs to be available else i need to replicate the logic locally eveytime and it's not safe (if you ever decide to change the way name is calculated, none of my GETs will work)

**Comment by benjaminbollen** [comment]
looks good to me; passing on to a second reviewer

**Pull request merged**